### PR TITLE
feat(httpd) allow overriding ingress and PV(C)s based on global value from parent chart

### PR DESCRIPTION
**Pull request by dduportal**
Same as https://github.com/jenkins-infra/helm-charts/pull/817 but for httpd

### CODE REVIEW

File: charts/httpd/tests/defaults_test.yaml [hunk 1]
```diff
@@ -5,30 +5,15 @@ templates:
   - ingress.yaml
   - secret.yaml # Direct dependency of deployment(.*).yaml
```
Comment (lemeurherve) [inline charts/httpd/tests/defaults_test.yaml:6]:
```suggestion
  - secret.yaml # Direct dependency of deployment.yaml
```
(nit; remnant of mirrorbits)

File: charts/httpd/tests/defaults_test.yaml [hunk 2]
```diff
@@ -68,17 +40,40 @@ tests:
           path: spec.template.metadata.labels["app.kubernetes.io/instance"]
           value: "RELEASE-NAME"
       - equal:
-          path: spec.template.spec.containers[0].imagePullPolicy
+          path: "spec.template.spec.containers[*].imagePullPolicy"
```
Comment (lemeurherve) [inline charts/httpd/tests/defaults_test.yaml:43]: Can there be more than one container? If not, I would have kept `0` to be like the other tests below. (nit)

**Review by lemeurherve — APPROVED** [review]

**Comment by lemeurherve** [comment]
Note: I've edited the PR title.

File: charts/httpd/tests/defaults_test.yaml (hunk 2 above)
Comment (dduportal) [reply charts/httpd/tests/defaults_test.yaml:43]: Yes there can be multiple containers (`spec.template.spec.containers` is a list).
The test here checks the imagepullpolicy set for any containers, it's the goal.

The test below have a different meaning and only checks the first container: it's not a matter of codestyle of homogeneity, but a matter of intent.

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**